### PR TITLE
[jsk_calibration] Write initial_poses.yaml to speed up optimization

### PR DESCRIPTION
**Pull request by garaemon**
@YoheiKakiuchi Please use this barnch to generate new motion.

### CODE REVIEW

**Comment by YoheiKakiuchi** [comment]
fix error, but I have no confidence around move-target.
~~~
diff --git a/jsk_calibration/euslisp/calibration.l b/jsk_calibration/euslisp/calibration.l
index 912b221..55b66c4 100644
--- a/jsk_calibration/euslisp/calibration.l
+++ b/jsk_calibration/euslisp/calibration.l
@@ -100,15 +100,18 @@
      (format f "repeat: False~%")
      ))
   (:write-initial-poses (strm)
-    (send robot-model :angle-vector angle-vector)
-    (let ((root (send robot-model :link root-link)))
-      (let ((trans (send (send root :copy-worldcoords) :transformation
-                         (send move-target :worldcoords))))
-        (let ((pos (send trans :worldpos))
-              (ypr (car (send trans :rpy-angle))))
-          (format f "- [~f, ~f, ~f, ~f, ~f, ~f]~%"
-                  (elt pos 0) (elt pos 1) (elt pos 2)
-                  (elt ypr 2) (elt ypr 1) (elt ypr 0))))))
+   (dolist (angle-vector motion-anglevectors)
+     (send robot-model :angle-vector angle-vector)
+     (let ((root (send robot-model :link root-link))
+           (move-target (send robot-model limb :end-coords :copy-worldcoords)))
+       (send move-target :transform end-effector-offset)
+       (let ((trans (send (send root :copy-worldcoords) :transformation
+                          (send move-target :worldcoords))))
+         (let ((pos (send trans :worldpos))
+               (ypr (car (send trans :rpy-angle))))
+           (format strm "- [~f, ~f, ~f, ~f, ~f, ~f]~%"
+                   (elt pos 0) (elt pos 1) (elt pos 2)
+                   (elt ypr 2) (elt ypr 1) (elt ypr 0)))))))
   (:generate-motion-yaml (root-directory arm angle-vector index tm)
     (let* ((fname (format nil "~A_~0,4d.yaml" (string arm) index))
            (full-path (format nil "~A/~A" root-directory fname)))
@@ -577,11 +580,11 @@
            (format f "      led_detector:~%")
            (format f "        active: False~%")
            )))))
-  (:generate-initial-poses (root-dir)
+  (:generate-initial-poses (root-directory)
     (let ((full-path (format nil "~A/capture_data/samples/initial_poses.yaml" root-directory)))
       (with-open-file (f full-path :direction :output)
         (dolist (motion motions)
-          (send motion :write-initial-poses f)
+          (send (cdr motion) :write-initial-poses f)
           )
         )))
   (:generate-files (root-directory)
~~~

**Comment by garaemon** [comment]
move-target should be a board object(cube).
I thought it was included in slots...

**Comment by garaemon** [comment]
Updated

File: jsk_calibration/euslisp/calibration.l [hunk 1]
```diff
@@ -354,8 +368,7 @@
     nil)
   (:check-collision ()
     (let ((result (send robot-model :self-collision-check :pairs collision-links)))
-      (pprint (list 'result result))
-      (if result (return-from :check-collision result))
+      (unless result (return-from :check-collision result))
```
Comment (YoheiKakiuchi) [inline jsk_calibration/euslisp/calibration.l]: Is this correct?
~~~
result -> nil (not collide) or collision pair (collide)
 if result -> return collision-pair
 unless result -> check robust collision
~~~

**Comment by garaemon** [comment]
hmm, looks wrong.

**Comment by garaemon** [comment]
it takes too much time, need to optimize it

**Comment by garaemon** [comment]
But now hrp2 has self collision check and it does not require this paranoia collision check.
staro has self collision check too?

**Comment by YoheiKakiuchi** [comment]
self collision check was processed once, because does not process robust collision check  https://github.com/jsk-ros-pkg/jsk_control/blob/master/jsk_calibration/euslisp/hrp2-calibration.l#L275

**Comment by garaemon** [comment]
Nop, I mentioned hrpsys

2015年1月14日水曜日、Yohei Kakiuchi<notifications@github.com>さんは書きました:

> self collision check was processed once, because does not process robust
> collision check
> https://github.com/jsk-ros-pkg/jsk_control/blob/master/jsk_calibration/euslisp/hrp2-calibration.l#L275
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_control/pull/194#issuecomment-69857172>
> .
>


-- 
from iPhone

**Comment by YoheiKakiuchi** [comment]
Not yet, but I think there is no problem to do so.

**Comment by garaemon** [comment]
Ok I will disable this paranoia collision check in default.Please add realtime-collision check to staro.


✉︎ Ryohei

**Pull request merged**